core/src/main/scala/org/apache/spark/sql/row/GemFireXDDialect.scala

```diff
@@ -116,12 +116,13 @@ abstract class GemFireXDBaseDialect extends JdbcExtendedDialect {
   override def getJDBCType(dt: DataType,
       md: Metadata): Option[JdbcType] = dt match {
     case StringType =>
-      if (md.contains("size")) {
-        if (md.contains("fixed") && md.getBoolean("fixed")) {
-          Some(JdbcType(s"CHAR(${md.getLong("size")})",
+      if (md.contains(Constant.CHAR_SIZE_NAME)) {
+        if (md.contains(Constant.CHAR_TYPE_NAME) &&
+            md.getString(Constant.CHAR_TYPE_NAME).equals("CHAR")) {
```

Comment:
Since since dialect mapping is there then we can actually store things as VARCHAR ot CHAR as the case may be instead of CLOB. I am not sure though. Please check. But we can do after 0.6